processor/src/main/java/com/linecorp/decaton/processor/processors/BatchingProcessor.java

```diff
@@ -0,0 +1,141 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.processors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.decaton.processor.DecatonProcessor;
+import com.linecorp.decaton.processor.DeferredCompletion;
+import com.linecorp.decaton.processor.ProcessingContext;
+import com.linecorp.decaton.processor.Completion;
+
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+/**
+ * An Abstract {@link DecatonProcessor} to Batch several tasks of type {@code T} to {@code List<T>}
+ * and process them at once. e.g. when downstream-DB supports batching I/O (which often very efficient).
+ * Batch-flushing should be done in time-based and size-based.
+ * @param <T> type of task to batch
+ */
+public abstract class BatchingProcessor<T> implements DecatonProcessor<T> {
+
+    private final ScheduledExecutorService executor;
+    private List<BatchingTask<T>> currentBatch = new ArrayList<>();
+    private final long lingerMillis;
+    private final int capacity;
+
+    @Getter
+    @Accessors(fluent = true)
+    public static class BatchingTask<T> {
+        @Getter
+        private final Completion completion;
+        @Getter
+        private final ProcessingContext<T> context;
+        @Getter
+        private final T task;
+
+        private BatchingTask(Completion completion, ProcessingContext<T> context, T task) {
+            this.completion = completion;
+            this.context = context;
+            this.task = task;
+        }
+    }
+
+    /**
+     * Instantiate {@link BatchingProcessor}.
+     * If you only need one limit, please set large enough value to another.
+     * @param lingerMillis time limit for this processor. On every lingerMillis milliseconds,
+     * tasks in past lingerMillis milliseconds are pushed to {@link BatchingTask#processBatchingTasks(List)}.
+     * @param capacity size limit for this processor. Every time tasks’size reaches capacity,
+     * tasks in past before reaching capacity are pushed to {@link BatchingTask#processBatchingTasks(List)}.
+     */
+    protected BatchingProcessor(long lingerMillis, int capacity) {
+        this.lingerMillis = lingerMillis;
+        this.capacity = capacity;
+
+        ScheduledThreadPoolExecutor scheduledExecutor = new ScheduledThreadPoolExecutor(1, r -> {
+            Thread th = new Thread(r);
```

Comment:
I think you can use `Utils.namedThreadFactory` for this.

Reply:
I fixed it. f6da8c6